include/xtensor/xjson.hpp

```diff
@@ -0,0 +1,153 @@
+            {
+                j = view();
+            }
+            else
+            {
+                j = nlohmann::json::array();
+                using size_type = typename D::size_type;
+                size_type nrows = view.shape()[0];
+                for (size_type i = 0; i != nrows; ++i)
+                {
+                    slices.push_back(i);
+                    nlohmann::json k;
+                    to_json_impl(k, e, slices);
+                    j.push_back(std::move(k));
+                    slices.pop_back();
+                }
+            }
+        }
+
+        template <class D>
+        inline void from_json_impl(const nlohmann::json& j, xexpression<D>& e, slice_vector& slices)
+        {
+            auto view = dynamic_view(e.derived_cast(), slices);
+
+            if (view.dimension() == 0)
+            {
+                view() = j;
+            }
+            else
+            {
+                using size_type = typename D::size_type;
+                size_type nrows = view.shape()[0];
+                for (auto i = 0; i != nrows; ++i)
+                {
+                    slices.push_back(i);
+                    const nlohmann::json& k = j[i];
+                    from_json_impl(k, e, slices);
+                    slices.pop_back();
+                }
+            }
+        }
+
+        inline unsigned int json_dimension(const nlohmann::json& j)
+        {
+            if (j.is_array() && j.size())
+            {
+                return 1 + json_dimension(j[0]);
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        template <class S>
+        inline void json_shape(const nlohmann::json& j, S& s, std::size_t pos = 0)
+        {
+            if (j.is_array())
+            {
+                auto size = j.size();
+                s[pos] = size;
+                if (size)
+                {
+                    json_shape(j[0], s, pos + 1);
+                }
+            }
+        }
+    }
+
+    template <class E>
+    inline enable_xexpression<E> to_json(nlohmann::json& j, const E& e)
+    {
+        auto sv = slice_vector();
+        detail::to_json_impl(j, e, sv);
+    }
+
+    template <class E>
+    inline enable_xcontainer<E> from_json(const nlohmann::json& j, E& e)
+    {
+        auto dimension = detail::json_dimension(j);
+        auto s = xtl::make_sequence<typename E::shape_type>(dimension);
+        detail::json_shape(j, s);
+
+        // In the case of a container, we resize the container.
+        e.resize(s);
+
+        auto sv = slice_vector();
+        detail::from_json_impl(j, e, sv);
+    }
+
+    template <class E>
+    inline enable_xview<E> from_json(const nlohmann::json& j, E& e)
+    {
+        typename E::shape_type s;
+        detail::json_shape(j, s);
+
+        // In the case of a view, we check the size of the container.
+        if (!std::equal(s.cbegin(), s.cend(), e.shape().cbegin()))
```

Comment:
maybe we should factor this out, like in `xt::resize_container` and make a `xt::resize_expression` free function, that resizes in the case of a container_semantic and throws if shape is wrong in case of view_semantic?
Could be useful for other cases in the future.